viewing_party/party.py

```diff
@@ -0,0 +1,168 @@
+    movie_genre_dict = {}
+    if len(user_data["watched"]) > 0:
+        for movie in user_data["watched"]:
+            genre = movie["genre"]
+            if genre not in movie_genre_dict:
+                movie_genre_dict[genre] = 1
+            else:
+                movie_genre_dict[genre] += 1
+    else:
+        return None
+
+    #loop over movie_genre_dict and have 2 temp variables and 
+    max_genre = 0
+    max_genre_key = ""
+    for genre, genre_freq in movie_genre_dict.items():
+        if genre_freq > max_genre:
+            max_genre_key = genre
+            max_genre = genre_freq
+
+    return max_genre_key
+
+    # UnboundLocalError: local variable 'max_genera' referenced before assignment
+    # look at each one to see which number is bigger
+    #need two variables most seen and number of times 
+    
+### WAVE 3 ###
+# determine which movies user has watched that listed friends have not watched.
+#Returns a list of dictionaries that represents a list of unique movies
+def get_unique_watched(user_data):
+    for item in user_data["watched"]:
+        print(item)
+    for item in user_data["friends"]:
+        print(item["watched"])
+        for watched_movie in item["watched"]:
+            print(user_data["watched"])
+            if watched_movie in user_data["watched"]:
+                user_data["watched"].remove(watched_movie)
+    return user_data["watched"]
+
+            
+def get_friends_unique_watched(user_data):
+    
+    unique_movies = []
+    
+    user_movies = set()
+    for watched_movie in user_data["watched"]:  
+        user_movies.add(watched_movie["title"])  
+        
+    friends_movies = set()    
+    for friend_data in user_data["friends"]:
+        for watched_movie in friend_data["watched"]:
+            friends_movies.add(watched_movie["title"])
+            
+    movies_unique_to_friends = friends_movies.difference(user_movies)
+    
+    for title in movies_unique_to_friends:
+        format_dict = {}
+        format_dict["title"] = title
+        unique_movies.append(format_dict)
+        
+    return unique_movies
+
+#### WAVE 4 ####
+        
+def get_available_recs(user_data):
+    #create list to store reccomendations 
+    recommendations = []
+    #create veriable to refer to subscriptions list
+    user_subscriptions = user_data["subscriptions"]
+    friend_recommendations = get_friends_unique_watched(user_data)
+    friends_movies = []
+    for movie in friend_recommendations:
+        friends_movies.append(movie["title"])
+    
+    #next find friend-recs with channels user is subscirbed to
+    for friend_data in user_data["friends"]:
+        for watched_movie_data in friend_data["watched"]:
+            #if that user is subsciebed to the host 
+            if watched_movie_data["host"] in user_subscriptions and watched_movie_data["title"] in friends_movies:
+                if watched_movie_data not in recommendations:
```

Comment:
Consider combining these into a compound conditional with two `and` to reduce nesting and enhance readability.